docs/core/task_library/contributing.md

```diff
@@ -0,0 +1,168 @@
+
+<script>
+import { Tweet } from 'vue-tweet-embed/dist'
+
+export default {
+    components: {Tweet}
+}
+</script>
+
+<Tweet id="1298298873878847490"></Tweet>
+
+## Task Structure
+
+In order to build a task for the task library you need to define the task's `__init__` and `run`
+functions. The `__init__` of the task will be called before the flow runs and the `run` function is where
+any of your task's logic will live to be executed at runtime. Allowing for kwargs to be set both during
+initialization and at runtime is key to improving a task's functionality.
+
+One example of this separation in action would be initializing a `ShellTask` with a specific `shell` and
+then passing in a different `command` at runtime. This will create two tasks in the flow, each with
+different commands, without having to redefine the shell type.
+
+```python
+my_shell = ShellTask(shell="bash")
+
+with Flow("shell_commands") as flow:
+    task1 = my_shell(command="ls")
+    task2 = my_shell(command="ls | wc -l")
+```
+
+(For more in depth information on the components of Prefect tasks take a look at
+[The Anatomy of a Prefect Task](/core/advanced_tutorials/task-guide.html) guide.)
+
+This snippet below is the general structure of a task contained in the task library.
+
+```python
+from prefect import Task
+from prefect.utilities.tasks import defaults_from_attrs
+
+class YourTask(Task):
+    """
+    Task for doing something
+
+    Args:
+        - your_kwarg (string, optional): an optional kwarg
+        - **kwargs: additional keyword arguments to pass to the Task constructor
+    """
+
+    def __init__(
+        self,
+        your_kwarg: str = None,
+        **kwargs: Any
+    ):
+        self.your_kwarg = your_kwarg
+        super().__init__(**kwargs)
+
+    @defaults_from_attrs("your_kwarg")
+    def run(self, your_kwarg: str = None) -> str:
+        """
+        Run your task
+
+        Args:
+            - your_kwarg (string, optional): an optional kwarg
+
+        Returns:
+            - string: description of what it returned
+        """
+
+        # Place your specific task run logic inside this function
+        return use_your_library(your_kwarg)
+```
+
+In the snippet above there is a special decorator `defaults_from_attrs`. This decorator serves the purpose
+of reducing the amount of boilerplate code in the task. If a value is set via initialization of the task
+and is not set again at runtime then the value set at initialization will be used in place of the absent
+runtime value. However, values set at runtime will always override those set during initialization.
+
+```python
+def __init__(self, your_kwarg: str = None, **kwargs: Any):
+    self.your_kwarg = your_kwarg
+    super().__init__(**kwargs)
+
+    @defaults_from_attrs("your_kwarg")
+    def run(self, your_kwarg: str = None) -> str:
+        use_your_library(your_kwarg)
+```
+
+is equivalent to
```

Comment:
Actually these aren't 100% equivalent - if a value is provided at runtime it is _always_ used regardless of the truthiness.  It can even be `None` (test for reference: https://github.com/PrefectHQ/prefect/blob/master/tests/utilities/test_tasks.py#L540-L546)

Reply:
Okay I'm actually going to remove these snippets